### PR TITLE
TW-3: Add top up by exolix and analytics

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/screens/buy/exolix/exchange-step.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      case 3:
+        event = ExolixSelectors.TopupThirdStepSupport;
+        break;
+      default:
+        event = ExolixSelectors.TopupFourthStepSubmit;
+        break;
+    }
+
+    return trackEvent(event, AnalyticsEventCategory.ButtonPress);
+  }, [step, trackEvent]);
```
Comment (LendiHop) [inline src/screens/buy/exolix/exchange-step.tsx:53]: and here

File: src/screens/buy/exolix/error-component.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      case 3:
+        event = ExolixSelectors.TopupThirdStepSupport;
+        break;
+      default:
+        event = ExolixSelectors.TopupFourthStepSubmit;
+        break;
+    }
+
+    return trackEvent(event, AnalyticsEventCategory.ButtonPress);
+  }, [step, trackEvent]);
```
Comment (LendiHop) [inline src/screens/buy/exolix/error-component.tsx:50]: mb create helper fn and reuse it here?

File: src/screens/buy/exolix/exolix-form-asset-input/exolix-asset-value-text.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { BigNumber } from 'bignumber.js';
+import React, { FC } from 'react';
+import { StyleProp, Text, TextStyle } from 'react-native';
+
+// import { CurrenciesInterface } from '../../../../interfaces/exolix.interface';
```
Comment (LendiHop) [inline src/screens/buy/exolix/exolix-form-asset-input/exolix-asset-value-text.tsx:5]: remove comment

File: src/screens/buy/exolix/exolix-form-asset-input/exolix-asset-value-text.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { isDefined } from '../../../../utils/is-defined';
+import { formatAssetAmount } from '../../../../utils/number.util';
+
+interface Props {
+  amount?: BigNumber;
+  // asset: CurrenciesInterface;
```
Comment (LendiHop) [inline src/screens/buy/exolix/exolix-form-asset-input/exolix-asset-value-text.tsx:11]: remove comment

File: src/screens/buy/exolix/use-topup-update.hook.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  const setExchangeData = useCallback(
+    (id: string) => {
+      dispatch(refreshExolixExchangeDataAction(id));
+    },
+    [dispatch]
+  );
+  const isAlive = useRef(false);
+
+  useEffect(() => {
+    let timeoutId = setTimeout(async function repeat() {
```
Comment (LendiHop) [inline src/screens/buy/exolix/use-topup-update.hook.ts:20]: why fn declaration?

File: src/store/exolix/exolix-epics.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import { combineEpics, Epic } from 'redux-observable';
+import { from, Observable, of } from 'rxjs';
+import { catchError, concatMap, map, switchMap } from 'rxjs/operators';
+import { Action } from 'ts-action';
+import { ofType, toPayload } from 'ts-action-operators';
+
+import {
+  // loadExolixCurrencies$,
```
Comment (LendiHop) [inline src/store/exolix/exolix-epics.ts:8]: remove comment

File: src/screens/buy/exolix/use-topup-update.hook.ts (hunk 5 above)
Comment (letier3110) [reply src/screens/buy/exolix/use-topup-update.hook.ts:20]: look below for another setTimeout calling after execution current timer: `timeoutId = setTimeout(repeat, 3000);`

File: src/store/exolix/exolix-epics.ts (hunk 6 above)
Comment (letier3110) [reply src/store/exolix/exolix-epics.ts]: look below for another setTimeout calling after execution current timer: timeoutId = setTimeout(repeat, 3000);

File: src/screens/buy/exolix/use-topup-update.hook.ts (hunk 5 above)
Comment (letier3110) [reply src/screens/buy/exolix/use-topup-update.hook.ts:20]: look below for another setTimeout calling after execution current timer: timeoutId = setTimeout(repeat, 3000);

**Review by LendiHop — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Pull request merged**